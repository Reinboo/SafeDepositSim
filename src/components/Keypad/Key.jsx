import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Key = ({ keyFace, handleClick }) => (
  <Wrapper type="button" id={keyFace} onClick={handleClick}>
    {keyFace}
  </Wrapper>
);

Key.propTypes = {
  keyFace: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

const Wrapper = styled.button`
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ${props => props.theme.buttonColor};
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ${props => props.theme.fontMedium};
  color: ${props => props.theme.buttonTextColor};
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ${props => props.theme.buttonColorPressed};
  }
`;

export default Key;
