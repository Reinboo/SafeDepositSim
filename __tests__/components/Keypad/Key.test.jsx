import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Key from '../../../src/components/Keypad/Key';


describe('Key component', () => {
  test('renders correctly', () => {
    const mockHandler = jest.fn();
    const wrapper = shallow(<Key keyFace="1" handleClick={mockHandler} />);
    const component = wrapper.dive();

    expect(toJson(component)).toMatchSnapshot();
  });

  test('invokes handler', () => {
    const mockHandler = jest.fn();
    const wrapper = mount(<Key keyFace="1" handleClick={mockHandler} />);
    wrapper.find('button').at(0).simulate('click');

    expect(mockHandler).toHaveBeenCalled();
  });
});
