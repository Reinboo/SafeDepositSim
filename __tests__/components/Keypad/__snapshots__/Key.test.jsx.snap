// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key component renders correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ",
          [Function],
          ";
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ",
          [Function],
          ";
  color: ",
          [Function],
          ";
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ",
          [Function],
          ";
  }
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  id="1"
  onClick={[MockFunction]}
  type="button"
>
  1
</StyledComponent>
`;
